frontend/purposeful-frontend/components/Registration.js

```diff
@@ -14,11 +14,93 @@ import {
 } from "@chakra-ui/react";
 import { ViewIcon, ViewOffIcon } from "@chakra-ui/icons";
 import { Field, Form, Formik } from "formik";
+import fetchWrapper, { getAuthorities } from "../utils/fetch_wrapper";
+import notification from "../utils/notification";
+
+async function handleRegistrationForm(values, actions, GrantedAuth) {
+  const payload = {
+    email: values.email,
+    password: values.password,
+    firstname: values.firstname,
+    lastname: values.lastname,
+  };
+  let response = null;
+
+  if (
+    GrantedAuth.length === 0 &&
+    !window.location.pathname.includes("moderator")
+  ) {
+    // User is not authenticated and is registering as a regular user
+    response = await fetchWrapper(
+      "/api/appuser/regular",
+      null,
+      "POST",
+      payload
+    );
+  } else if (
+    GrantedAuth.includes("Owner") &&
+    window.location.pathname.includes("moderator")
+  ) {
+    // User is an admin and is registering a moderator
+    response = await fetchWrapper(
+      "/api/appuser/moderator",
+      null,
+      "POST",
+      payload
+    );
+  } else {
+    // User does not have the required permissions
+    notification(
+      "error",
+      "You do not have the required permissions to perform this action.",
+      null
+    );
+    actions.setSubmitting(false);
+  }
+
+  if (
+    response.ok &&
+    GrantedAuth.includes("Owner") &&
+    window.location.pathname.includes("moderator")
+  ) {
+    // User is an admin and successfully registered a moderator
+    notification("success", "Moderator account created successfully.", null);
```

Comment:
Here is where the notification popup component is used ( as a function )? Very nice

Reply:
Correct